web-search-engines/langchain4j-web-search-engine-searchapi/src/main/java/dev/langchain4j/web/search/searchapi/SearchApiClient.java

```diff
@@ -0,0 +1,72 @@
+package dev.langchain4j.web.search.searchapi;
+
+import static com.google.gson.FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES;
+
+import java.io.IOException;
+import java.time.Duration;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import lombok.Builder;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class SearchApiClient {
+	private final SearchApiAPI searchapiAPI;
+	private final String engine;
+	private final String apiKey;
+
+	/* Indicates whether to log requests and their responses. Defaults to false. */
+	private final boolean logRequests;
+
+	private static final Gson GSON = new GsonBuilder().setFieldNamingPolicy(LOWER_CASE_WITH_UNDERSCORES)
+			.setPrettyPrinting().create();
+
+	@Builder
+	public SearchApiClient(final String baseUrl, final String apiKey, final String engine, final Duration timeout,
+			final boolean logRequests) {
```

Comment:
`logResponses` is missing, is it on purpose?

Reply:
Yes, it's on purpose.

The `langchain4j-web-search-engine-google-custom` module uses [two separate](https://github.com/langchain4j/langchain4j/blob/d23b67567f64c9a2a203704370679460ec92fcce/web-search-engines/langchain4j-web-search-engine-google-custom/src/main/java/dev/langchain4j/web/search/google/customsearch/GoogleCustomSearchApiClient.java#L35-L36) flags i.e. `logRequests` and `logResponses` because the HTTP client it uses [supports two individual](https://github.com/langchain4j/langchain4j/blob/d23b67567f64c9a2a203704370679460ec92fcce/web-search-engines/langchain4j-web-search-engine-google-custom/src/main/java/dev/langchain4j/web/search/google/customsearch/GoogleCustomSearchApiClient.java#L57-L61) flags to enable logging of requests and responses.

My implementation is based on the `langchain4j-web-search-engine-tavily` module but the developer did not enable [support for logging](https://github.com/langchain4j/langchain4j/blob/d23b67567f64c9a2a203704370679460ec92fcce/web-search-engines/langchain4j-web-search-engine-tavily/src/main/java/dev/langchain4j/web/search/tavily/TavilyClient.java#L26). When I added [support for logging](https://github.com/ayewo/langchain4j/blob/2701180c06468bfd930e67d0863e2a573bc9e850/web-search-engines/langchain4j-web-search-engine-searchapi/src/main/java/dev/langchain4j/web/search/searchapi/SearchApiClient.java#L39-L43) in my implementation, I realized that the HTTP client (`OkHttpClient`) only accepts one flag to enable logging which is why I removed `logResponses` since it was redundant.